engine/context/src/main/java/io/deephaven/engine/context/ExecutionContext.java

```diff
@@ -164,14 +182,29 @@ public QueryCompiler getQueryCompiler() {
         return queryCompiler;
     }
 
+    public AuthContext getAuthContext() {
+        return authContext;
+    }
+
     @SuppressWarnings("unused")
     public static class Builder {
         private boolean isSystemic = false;
 
+        private final AuthContext authContext;
+
         private QueryLibrary queryLibrary = PoisonedQueryLibrary.INSTANCE;
         private QueryScope queryScope = PoisonedQueryScope.INSTANCE;
         private QueryCompiler queryCompiler = PoisonedQueryCompiler.INSTANCE;
 
+        private Builder() {
+            // propagate the auth context from the current context
+            this(getContext().authContext);
```

Comment:
This is a bit of a departure from how the other parts are "poisened". Do we think that callers should be forced to do something like `captureAuthContext()` instead?

Reply:
How much cooperation is appropriate to ask of the user(s)? I feel that without being able to offer an alternative (as that feels like a security faux pas) that passing along the current value feels friendly and helpful.